R/in-notin.R

```diff
@@ -22,14 +25,21 @@
 '%in%' <- function(x, y) {
   if (is.character(x) && is.character(y)) {
     x %chin% y
-  } else if (is.list(y)) {
-    # https://github.com/markfairbanks/tidytable/issues/565
+  } else if (needs_base_in(x, y)) {
```

Comment:
As I think about this more - I think it makes more sense that `base::'%in%'` is the "fallback" else option at the end.

``` r
'%in%' <- function(x, y) {
  if (is.character(x) && is.character(y)) {
    x %chin% y
  } else if (vec_ptype_compatible(x, y)) {
    vec_in(x, y)
  } else {
    base::'%in%'(x, y)
  }
}

# Check if two vectors have compatible ptypes
vec_ptype_compatible <- function(x, y) {
  # https://github.com/markfairbanks/tidytable/issues/565
  # https://github.com/markfairbanks/tidytable/issues/632
  tryCatch({vec_ptype_common(x, y); TRUE}, error = function(e) FALSE)
}
```